src/core.c/IO/Path.pm6

```diff
@@ -682,10 +684,11 @@ my class IO::Path is Cool does IO {
         # check if we need a BOM
         if $enc eq 'utf16' {
 
+           my $stat := nqp::dispatch("boot-syscall", "file-stat", nqp::decont_s($path), 0);
            # add a BOM if writing starts at beginning of file
            if $mode eq 'w'
```

Comment:
Could probably rewrite this conditional so that the `stat` doesn't occur unless `$mode eq 'w'`

Reply:
Implemented in [181921e](https://github.com/rakudo/rakudo/pull/5316/commits/181921e0582e327d1bfe089836aff90397d6b985). As is it requires repeating the actual BOM writing logic, but if this is deemed to violate DRY, it could be changed to just set a flag.